iterreg/sparse/solvers.py

```diff
@@ -254,3 +254,42 @@ def fista(X, y, alpha, prox=shrink, pen=ell1, max_iter=1_000, f_store=10,
             if verbose:
                 print(t, E[t // f_store])
     return w, all_w, E
+
+
+@njit
+def adaptive(X, y, alpha, prox=shrink, pen=ell1, max_iter=1_000, n_adapt=2,
```

Comment:
prox should not be needed, instead a reweighting function (giving the value of the derivative of the penalty at w[j])